kanidmd/src/lib/be/idl_sqlite.rs

```diff
@@ -220,153 +212,122 @@ pub trait IdlSqliteTransaction {
     }
 
     // ! TRACING INTEGRATED
     fn get_idl(
         &self,
-        audit: &mut AuditScope,
         attr: &str,
         itype: &IndexType,
         idx_key: &str,
     ) -> Result<Option<IDLBitRange>, OperationError> {
         spanned!("be::idl_sqlite::get_idl", {
-            lperf_trace_segment!(audit, "be::idl_sqlite::get_idl", || {
-                if !(self.exists_idx(audit, attr, itype)?) {
-                    filter_error!("Index {:?} {:?} not found", itype, attr);
-                    lfilter_error!(audit, "Index {:?} {:?} not found", itype, attr);
-                    return Ok(None);
-                }
-                // The table exists - lets now get the actual index itself.
+            if !(self.exists_idx(attr, itype)?) {
+                filter_error!("Index {:?} {:?} not found", itype, attr);
+                return Ok(None);
+            }
+            // The table exists - lets now get the actual index itself.
 
-                let query = format!(
-                    "SELECT idl FROM idx_{}_{} WHERE key = :idx_key",
-                    itype.as_idx_str(),
-                    attr
-                );
-                let mut stmt = self
-                    .get_conn()
-                    .prepare(query.as_str())
-                    .map_err(sqlite_error)?;
-                let idl_raw: Option<Vec<u8>> = stmt
-                    .query_row(&[(":idx_key", &idx_key)], |row| row.get(0))
-                    // We don't mind if it doesn't exist
-                    .optional()
-                    .map_err(sqlite_error)?;
+            let query = format!(
+                "SELECT idl FROM idx_{}_{} WHERE key = :idx_key",
+                itype.as_idx_str(),
+                attr
+            );
+            let mut stmt = self
+                .get_conn()
+                .prepare(query.as_str())
+                .map_err(sqlite_error)?;
+            let idl_raw: Option<Vec<u8>> = stmt
+                .query_row(&[(":idx_key", &idx_key)], |row| row.get(0))
+                // We don't mind if it doesn't exist
+                .optional()
+                .map_err(sqlite_error)?;
 
-                let idl = match idl_raw {
-                    Some(d) => serde_cbor::from_slice(d.as_slice()).map_err(serde_cbor_error)?,
-                    // We don't have this value, it must be empty (or we
-                    // have a corrupted index .....
-                    None => IDLBitRange::new(),
-                };
-                trace!(%idl, "Got idl for index {:?} {:?}", itype, attr);
-                ltrace!(audit, "Got idl for index {:?} {:?} -> {}", itype, attr, idl);
+            let idl = match idl_raw {
+                Some(d) => serde_cbor::from_slice(d.as_slice()).map_err(serde_cbor_error)?,
+                // We don't have this value, it must be empty (or we
+                // have a corrupted index .....
+                None => IDLBitRange::new(),
+            };
```

Comment:
Use if-let?

Reply:
Probably should be unwrap_on_else, but see above re logic change.